libs/datamodel/connectors/datamodel-connector/src/connector_error.rs

```diff
@@ -132,6 +139,12 @@ pub enum ErrorKind {
         connector_name: String,
     },
 
+    #[error("Native type {} can not be used in an ID field in {}.", native_type, connector_name)]
```

Comment:
Seems a bit odd to capitalize `id`. If that looks weird to you maybe adapt it to something like:
```
Native type {} of {} can not be used on a field that is `@id` or `@@id`.
```

Reply:
Yes good point, that sounds a lot better.